ntbs-service/Models/Entities/SocialContextAddress.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+using ntbs_service.Models.Validations;
+
+namespace ntbs_service.Models.Entities
+{
+    public class SocialContextAddress : SocialContextBase
+    {
+        public int SocialContextAddressId { get; set; }
+
+        [Required(ErrorMessage = ValidationMessages.RequiredEnter)]
+        [RegularExpression(ValidationRegexes.PostcodeValidation, ErrorMessage = ValidationMessages.PostcodeIsNotValid)]
+        [DisplayName("Postcode")]
+        public string Postcode { get; set; }
+
+        public override void SetModelId(int id)

```

Comment:
What's this needed for?

Reply:
@jasiekmiko In the socialcontextbase razor page model, we need to set the id for an entity of type socialcontextbase - as the Id has a different name for each, we need a different implementation of setting this.
An alternative would be to use a property instead of a method - i.e. 
`public abstract int Id { set; }`
in SocialContextBase and then 
` public override int Id
        {
            set {
                SocialContextAddressId = value;
            }
        }`
in Address etc for Venue.
Then in the razor page we could call `model.Id = RowId.Value`, rather than the method. Do you think that's clearer?

Reply:
Ooh, yeah, using a setter seems really nice actually!

Reply:
I thought so too, so decided to change to setter (if you check the most recent commit)